### DISCUSSION TITLE
Continuously checking botocore using GH Actions

### DISCUSSION THREAD

**Discussion by Andrew-Chen-Wang** [aio-libs/aiobotocore#843] [Ideas]
Hey, neat package! I saw in a different issue about the pinning of botocore. I see you were worried about things breaking. One way we could potentially fix this is by versioning aiobotocore with botocore. Something like mypy-boto3's CI/CD is doing.

Otherwise, a different idea to continuously check if aiobotocore supports a certain botocore version is to do a daily CI run using GitHub actions. We could update a certain MD file to show which versions of botocore is supported by which *latest* version aiobotocore (hope that made sense).

**Comment by thehesiod** [comment]
the problem is there's still a chicken an egg problem, each version of botocore could introduce new features, which means we won't have tests for these new features.  We'd have to automate running botocore's tests asynchronously as well which would be quite the undertaking :)

**Reply by Andrew-Chen-Wang** [reply]
When you mean running botocore's tests asynchronously, do you mean converting them to an async context? Not sure if you mean something like `asgiref.sync_to_async`, but I'm pretty sure you mean just running new features with the aiobotocore client and session... definitely an issue.

I was mainly pointing out how to test existing code in aiobotocore, but for sure new features in botocore need to be checked as well. I've never seen a CI do -- uh -- that before; mostly just people checking the release/changelog...

**Reply by thehesiod** [reply]
ya, programmatically converting their sync tests to async, should't be too hard as you could check if the object returned is a coroutine and loop.run_until_complete on it, would probably cover ~80% of the tests, for sure there would be complicated issues though.  That's the only way that would ensure that the new botocore works with the existing aiobotocore...but that still relies on them having perfect coverage as well :). Nothing matches a manual diff across the two versions and applying said changes to aiobotocore for now.  We need a human to make that decision for now.  Another option is vendoring but that is painful and would really bloat this project.

ya we could automate testing existing aiobotocore with the latest botocore but that doesn't help much as you couldn't take any action on that result because it would not guarantee that not only existing APIs, nor new APIs would work
